File: php/src/Resources/PeppolDirectory.php
<?php

declare(strict_types=1);

namespace EPostak\Resources;

use EPostak\HttpClient;
use EPostak\EPostakError;

/**
 * Search the Peppol Business Card directory.
 *
 * Access via `$client->peppol->directory`.
 */
class PeppolDirectory
{
    /**
     * @param HttpClient $http Shared HTTP transport instance.
     */
    public function __construct(private HttpClient $http)
    {
    }

    /**
     * Search the Peppol Business Card directory.
     *
     * Requires `documents:read` scope.
     *
     * @param array{q?: string, country?: string, page?: int, page_size?: int} $params Search filters:
     *   - `q`         Free-text search query (company name, ID, etc.).
     *   - `country`   ISO 3166-1 alpha-2 country code (e.g. 'SK', 'CZ').
     *   - `page`      Page number (1-based).
     *   - `page_size` Results per page (default 20).
     * @return array Paginated search results with matching business card entries.
     * @throws EPostakError On API error.
     */
    public function search(array $params = []): array
    {
        $qs = HttpClient::buildQuery([
            'q' => $params['q'] ?? null,
            'country' => $params['country'] ?? null,
            'page' => $params['page'] ?? null,
            'page_size' => $params['page_size'] ?? null,
        ]);
        return $this->http->request('GET', '/peppol/directory/search' . $qs);
    }
}

File: ruby/lib/epostak/resources/peppol_directory.rb
# frozen_string_literal: true

module EPostak
  module Resources
    # Sub-resource for searching the Peppol Business Card directory.
    # The directory contains publicly registered Peppol participants.
    #
    # Accessed via +client.peppol.directory+.
    #
    # @example
    #   results = client.peppol.directory.search(q: "consulting", country: "SK")
    #   results["results"].each { |r| puts r["name"] }
    class PeppolDirectory
      # @param http [EPostak::HttpClient] Internal HTTP client
      def initialize(http)
        @http = http
      end

      # Search the Peppol Business Card directory for registered participants.
      # Supports free-text search and country filtering with pagination.
      #
      # Requires +documents:read+ scope.
      #
      # @param q [String, nil] Free-text search query (company name, ID, etc.)
      # @param country [String, nil] ISO 3166-1 alpha-2 country code filter (e.g. "SK", "CZ")
      # @param page [Integer, nil] Page number (1-based)
      # @param page_size [Integer, nil] Results per page
      # @return [Hash] Paginated search results with "results" array and "total" count
      #
      # @example Search for Slovak companies
      #   results = client.peppol.directory.search(q: "consulting", country: "SK", page_size: 50)
      #   puts "Found #{results['total']} participants"
      def search(q: nil, country: nil, page: nil, page_size: nil)
        query = { q: q, country: country, page: page, page_size: page_size }
        @http.request(:get, "/peppol/directory/search", query: query)
      end
    end
  end
end
